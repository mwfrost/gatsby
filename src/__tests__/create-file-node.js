const path = require(`path`)

const { createFileNode } = require(`../create-file-node`)

describe(`create-file-node`, () => {
  it(`creates a file node`, done => {
    createFileNode(
      path.resolve(`${__dirname}/fixtures/file.json`),
      {},
      (err, fileNode) => {
        // Delete access time since this changes on every run.
        delete fileNode.accessTime
        delete fileNode.atime
        delete fileNode.atimeMs

        expect(fileNode).toMatchSnapshot()
        done()
      }
    )
  })
})
