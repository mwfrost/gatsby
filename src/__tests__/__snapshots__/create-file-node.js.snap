// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-file-node creates a file node 1`] = `
Object {
  "absolutePath": "/Users/kylemathews/programs/gatsby/packages/gatsby-source-filesystem/src/__tests__/fixtures/file.json",
  "base": "file.json",
  "birthTime": "2017-08-26T02:16:56.000Z",
  "birthtime": "2017-08-26T02:16:56.000Z",
  "birthtimeMs": 1503713816000,
  "blksize": 4096,
  "blocks": 8,
  "changeTime": "2017-08-26T02:16:56.000Z",
  "children": Array [],
  "ctime": "2017-08-26T02:16:56.000Z",
  "ctimeMs": 1503713816000,
  "dev": 16777220,
  "dir": "/Users/kylemathews/programs/gatsby/packages/gatsby-source-filesystem/src/__tests__/fixtures",
  "ext": ".json",
  "extension": "json",
  "gid": 20,
  "id": "/Users/kylemathews/programs/gatsby/packages/gatsby-source-filesystem/src/__tests__/fixtures/file.json absPath of file",
  "ino": 140178047,
  "internal": Object {
    "contentDigest": "3857af536a3d0950f833cf47079facdd",
    "mediaType": "application/json",
    "type": "File",
  },
  "mode": 33188,
  "modifiedTime": "2017-08-26T02:16:56.000Z",
  "mtime": "2017-08-26T02:16:56.000Z",
  "mtimeMs": 1503713816000,
  "name": "file",
  "nlink": 1,
  "parent": "___SOURCE___",
  "prettySize": "14 B",
  "rdev": 0,
  "relativePath": "src/__tests__/fixtures/file.json",
  "root": "/",
  "size": 14,
  "sourceInstanceName": "__PROGRAMATTIC__",
  "uid": 501,
}
`;
